### PR TITLE
Bazelize bazel-bench

### PR DESCRIPTION
**Pull request by joeleba**
**What this PR does and why we need it:**

This PR enables the use of Bazel to build & test `bazel-bench`.

**New changes / Issues that this PR fixes:**

- Added various build/test targets.
- Updated README to reflect this big change.

**Special notes for reviewer:**

If possible, try to clone and set up the project from scratch.

**Does this require a change in the script's interface or the BigQuery's table structure?**

No

### CODE REVIEW

File: BUILD [hunk 1]
```diff
@@ -0,0 +1,20 @@
+load("@third_party//:requirements.bzl", "requirement")
+
+py_binary(
+    name = "benchmark",
+    srcs = ["benchmark.py"],
+    deps = [
+        "//utils:utils",
+      	requirement('absl-py'),
```
Comment (meisterT) [inline BUILD]: the indentation seems to be wrong

File: BUILD [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+        requirement('GitPython'),
+    ],
+    legacy_create_init = 0
```
Comment (meisterT) [inline BUILD]: does it break if you remove this?

File: README.md [hunk 3]
```diff
@@ -11,11 +11,27 @@ Pre-requisites: `python`, `pip`, `git`, `bazel`
 To do a test run:
 
 1.  The use of `virtualenv` is strongly recommended. Do this before you carry on
```
Comment (meisterT) [inline README.md]: I didn't do all these virtualenv steps. The build still succeeded. Is that the reason for my test failures?

File: WORKSPACE [hunk 4]
```diff
@@ -0,0 +1,25 @@
+load("@bazel_tools//tools/build_defs/repo:git.bzl", "git_repository")
+
+git_repository(
+    name = "io_bazel_rules_python",
+    remote = "https://github.com/bazelbuild/rules_python.git",
+    # NOT VALID!  Replace this with a Git commit SHA.
```
Comment (meisterT) [inline WORKSPACE]: :question:

File: utils/BUILD [hunk 5]
```diff
@@ -0,0 +1,72 @@
...
+      ["*.py"],
+      exclude = ["*_test.py"]
+  )
+)
+
+py_library(
+  name = "utils",
+  srcs = [":utils-srcs"],
+  deps = [
+      requirement('cachetools'),
```
Comment (meisterT) [inline utils/BUILD]: wow, that's a long list ;-)

File: utils/bazel_test.py [hunk 6]
```diff
@@ -57,7 +57,8 @@ def test_get_times(self, process_mock, unused_time_mock, unused_get_pid_mock):
   @mock.patch.object(bazel.Bazel, '_get_heap_size')
   @mock.patch.object(bazel.Bazel, '_get_times')
   @mock.patch.object(bazel.subprocess, 'Popen')
-  def test_command(self, subprocess_mock, get_times_mock, get_heap_size_mock,
+  @mock.patch('datetime.datetime')
```
Comment (meisterT) [inline utils/bazel_test.py]: how are the changes to this file related to the PR?

**Comment by Globegitter** [comment]
Not sure if these `BUILD` files where generated by hand, but I had success using https://github.com/tuomasr/pazel to generate optimal `BUILD` files for me (sometimes with a little bit of hand tweaking afterwards). Also there are alternative rules to interact with pip: https://github.com/apt-itude/rules_pip. They support sub-dependency locking out of the box, could be worth giving a shot. Anyway, just my two cents, feel free to ignore.

File: utils/bazel_test.py (hunk 6 above)
Comment (joeleba) [reply utils/bazel_test.py]: Moved that to PR #14

**Comment by joeleba** [comment]
> Not sure if these `BUILD` files where generated by hand, but I had success using https://github.com/tuomasr/pazel to generate optimal `BUILD` files for me (sometimes with a little bit of hand tweaking afterwards). Also there are alternative rules to interact with pip: https://github.com/apt-itude/rules_pip. They support sub-dependency locking out of the box, could be worth giving a shot. Anyway, just my two cents, feel free to ignore.

@Globegitter thanks for the recommendation! Yes, those were generated by hand.

File: utils/output_handling.py [hunk 7]
```diff
@@ -73,6 +72,8 @@ def upload_csv(csv_file_path, bigquery_cfg):
     bigquery_cfg: The string representing the BigQuery table config. Comes in
       the form <dataset_id>:<table_id>:<location>
   """
+  from google.cloud import bigquery
```
Comment (meisterT) [inline utils/output_handling.py]: What's the motivation to move this import down here?

Comment (joeleba) [reply utils/output_handling.py]: It's a workaround of https://github.com/bazelbuild/rules_python/issues/14 for `bazel test` to work.

**Pull request merged**